pkg/agent/flowexporter/exporter/exporter.go

```diff
@@ -539,3 +553,34 @@ func (exp *flowExporter) sendDataSet() (int, error) {
 	klog.V(4).Infof("Data set sent successfully. Bytes sent: %d", sentBytes)
 	return sentBytes, nil
 }
+
+func (exp *flowExporter) findFlowType(record flowexporter.FlowRecord) uint8 {
+	if record.Conn.Mark == openflow.ServiceCTMark || exp.ipInPodSubnets(record.Conn.TupleOrig.SourceAddress) && exp.ipInPodSubnets(record.Conn.TupleOrig.DestinationAddress) {
+		if record.Conn.SourcePodName == "" || record.Conn.DestinationPodName == "" {
+			return ipfixregistry.InterNode
+		}
+		return ipfixregistry.IntraNode
+	} else {
+		// Update Pod subnets to distinguish Pod-To-External flows.
+		if exp.nodeRouteController != nil {
+			exp.podSubnets = exp.nodeRouteController.GetPodSubnetsFromAllNodes()
```

Comment:
Do we call it for every flow? Then we should optimize here; probably by maintaining a single map in NodeRouteController and look up there, or even require a podCIDR parameter to add flow type.

Reply:
We call it if a flow record is not a Pod-To-Pod or Pod-To-Service record. Since new nodes may be added into the cluster anytime, current solution is that if we failed to judge current record is a Pod-To-Pod or Pod-To-Service record, we will fetch the latest PodCIDRs from NodeRouteController, then distinguish if it is a Pod-To-External flow.

Reply:
But when this happens - failed to judge current record is a Pod-To-Pod or Pod-To-Service record?

Reply:
To be brief, if the destination IP of a record is not in all PodSubnets we currently known.

Reply:
Does it mean every flow to external will end up here?

Reply:
Yes

Reply:
Then we should optimize in my mind.

Reply:
Agreed. @dreamtalen Please optimize this as discussed offline.

Reply:
Thanks, updated PR with following optimization:
Maintained a PodCIDRIPsMap in NodeRouteController to avoid looking up PodSubsets in flow exporter when send record.
Took advantage of the fixed mask size among PodCIDRs, using Map to accelerate the judgement of IP in PodSubnets.